src/robot/libraries/Collections.py

```diff
@@ -73,10 +93,9 @@ def insert_into_list(self, list_, index, value):
         | ${L1} = ['xxx', 'a']
         | ${L2} = ['a', 'xxx', 'b']
         """
-        self._validate_list(list_)
-        list_.insert(self._index_to_int(index), value)
+        list_.insert(index, value)
 
-    def combine_lists(self, *lists):
+    def combine_lists(self, *lists: list) -> list:
```

Comment:
Input type could be `Sequence`. It doesn't matter in practice because `chain.from_iterable` works the same way with both types, but conceptually there's no need to require lists. If we think about possible Python usage, the current typing disallows `combine_lists([1, 2], (3, 4))` which is actually supported.